backend/reviewduck/src/main/java/com/reviewduck/review/repository/ReviewRepository.java

```diff
@@ -26,5 +27,9 @@ public interface ReviewRepository extends Repository<Review, Long> {
 
     Page<Review> findByMemberAndIsPrivateFalse(Member member, Pageable pageable);
 
+    @Modifying(clearAutomatically = true)
```

Comment:
이 어노테이션을 붙인 이유는 무엇인가요?

Reply:
이 어노테이션이 붙은 메서드를 실행시 영속성 컨텍스트를 비워주기 위함입니다. 이유는 좋아요를 더한 경우 갱신된 최신 좋아요 개수를 반환해줘야 하는데 고려한 방법은 다음과 같습니다. 

1. EntityManager를 주입받아 refresh 메서드를 사용하여 DB -> 영속성 컨텍스트의 방향으로 동기화시킨다. 
2. 수정 메서드와 조회 메서드를 분리하여 개별 트랜잭션에서 작업한다.
3. 수정 메서드 실행시 영속성 컨텍스트를 비우고 같은 트랜잭션에서 조회 메서드로 최신 값을 불러온다.

이 방안 중에서 1번은 이 메서드만을 위해 다른 Service에서도 주입받지 않은 EntityManager를 주입받는 게 혼동의 여지도 있고 깔끔하다고 생각하지 않아 보류하였습니다. 
2번은 불필요한 트랜잭션을 새롭게 수행하는 것으로 보아 보류하였습니다. 
따라서 3번을 방안을 선택하였고, 이를 위해 메서드에 clearAutomatically 옵션을 true로 주어 영속성 컨텍스트를 비우고 조회하여 한 트랜잭션 내에서 DB에 갱신된 최신 값을 읽어오도록 하였습니다. 

아래는 참고입니다. 

If our modifying query changes entities contained in the persistence context, then this context becomes outdated. One way to manage this situation is to [clear the persistence context](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#clear--). By doing that, we make sure that the persistence context will fetch the entities from the database next time.

However, we don't have to explicitly call the clear() method on the EntityManager. We can just use the [clearAutomatically property](https://codingexplained.com/coding/java/spring-framework/updating-entities-with-update-query-spring-data-jpa) from the @Modifying annotation:

@Modifying(clearAutomatically = true)
This way, we make sure that the persistence context is cleared after our query execution.

However, if our persistence context contained unflushed changes, clearing it would mean dropping the unsaved changes. Fortunately, there's another property of the annotation we can use in this case, flushAutomatically:

@Modifying(flushAutomatically = true)
Now the EntityManager is flushed before our query is executed.

[Spring Data JPA @Modifying Annotation](https://www.baeldung.com/spring-data-jpa-modifying-annotation)